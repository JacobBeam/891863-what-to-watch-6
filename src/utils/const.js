export const TabType = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const RatingLevels = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`
};
