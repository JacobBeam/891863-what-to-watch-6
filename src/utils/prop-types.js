import PropTypes from 'prop-types';

export const filmsPropTypes = {
  films: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    posterImage: PropTypes.string.isRequired,
    previewImage: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    runTime: PropTypes.number.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    id: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    videoLink: PropTypes.string.isRequired,
    previewVideoLink: PropTypes.string.isRequired,
  })).isRequired,
};

export const filmPropTypes = {
  film: PropTypes.shape({
    name: PropTypes.string.isRequired,
    posterImage: PropTypes.string.isRequired,
    previewImage: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    runTime: PropTypes.number.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    id: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    videoLink: PropTypes.string.isRequired,
    previewVideoLink: PropTypes.string.isRequired,
  }).isRequired,
};
