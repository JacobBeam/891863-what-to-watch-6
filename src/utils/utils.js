import {RatingLevels, allGenreFilter} from './const';

export const findFilmById = (films, id) => {
  return films.find((element) => element.id === id);
};


export const conversionMinutes = (mins) => {
  const hours = Math.trunc(mins / 60);
  const minutes = mins % 60;
  return `${hours}h ${minutes}m`;
};


export const findRatingLevel = (rating) => {

  let currentRating = ``;

  switch (true) {
    case (rating >= 0 && rating < 3):
      currentRating = RatingLevels.BAD;
      break;
    case (rating >= 3 && rating < 5):
      currentRating = RatingLevels.NORMAL;
      break;
    case (rating >= 5 && rating < 8):
      currentRating = RatingLevels.GOOD;
      break;
    case (rating >= 5 && rating < 10):
      currentRating = RatingLevels.VERY_GOOD;
      break;
    case (rating >= 10):
      currentRating = RatingLevels.AWESOME;
      break;
  }
  return currentRating;
};

export const filterFilmsByGenre = (genre, films) => {

  if (genre === allGenreFilter.ALL_GENRES) {
    return films;
  }

  return films.filter((film)=>film.genre === genre);
};
