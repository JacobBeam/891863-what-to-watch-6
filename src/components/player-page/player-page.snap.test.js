import React from 'react';
import {render} from '@testing-library/react';
import {createMemoryHistory} from 'history';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {Router} from 'react-router-dom';
import PlayerPage from './player-page';
import {AuthorizationStatus} from '../../utils/const';

const api = jest.fn();
const middlleware = thunk.withExtraArgument(api);
const mockStore = configureStore([middlleware]);
const onFollowingGoBack = jest.fn();
const defaultProps = {
  match: {params: {id: `0`}},
};
it(`Should PlayerPage render correctly`, () => {
  const history = createMemoryHistory();
  const store = mockStore({
    USER: {authorizationStatus: AuthorizationStatus.AUTH},
    FILM_DATA: {
      selectedMovie: {
        "name": `Macbeth`,
        "backgroundImage": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
        "videoLink": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
      },
      isSelectedFilmLoaded: true}
  });

  const {container} = render(
      <redux.Provider store={store}>
        <Router history={history}>
          <PlayerPage {...defaultProps} onFollowingGoBack={onFollowingGoBack}/>
        </Router>
      </redux.Provider>
  );
  expect(container).toMatchSnapshot();
});
