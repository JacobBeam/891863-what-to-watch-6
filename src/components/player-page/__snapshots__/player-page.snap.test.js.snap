// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PlayerPage render correctly 1`] = `
<div>
  <div
    class="player"
  >
    <video
      class="player__video"
      poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg"
      src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
    />
    <button
      class="player__exit"
      data-testid="exit"
      type="button"
    >
      Exit
    </button>
    <div
      class="player__controls"
    >
      <div
        class="player__controls-row"
      >
        <div
          class="player__time"
        >
          <progress
            class="player__progress"
            max="100"
            value="0"
          />
          <div
            class="player__toggler"
            style="left: 0%;"
          >
            Toggler
          </div>
        </div>
        <div
          class="player__time-value"
        >
          00:00:00
        </div>
      </div>
      <div
        class="player__controls-row"
      >
        <button
          class="player__play"
          data-testid="play"
          type="button"
        >
          <svg
            height="19"
            viewBox="0 0 19 19"
            width="19"
          >
            <use
              xlink:href="#play-s"
            />
          </svg>
          <span>
            Play
          </span>
        </button>
        <div
          class="player__name"
        >
          Macbeth
        </div>
        <button
          class="player__full-screen"
          data-testid="full-screen"
          type="button"
        >
          <svg
            height="27"
            viewBox="0 0 27 27"
            width="27"
          >
            <use
              xlink:href="#full-screen"
            />
          </svg>
          <span>
            Full screen
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
